scripts/coccinelle/identifier_length.cocci

```diff
@@ -3,6 +3,20 @@
 
 virtual report
 
+@initialize:python
+  depends on report
+@
+@@
+identifiers = {}
```

Comment:
Am I correct, that this dictionary will store all identifiers from whole Zephyr during build?

Reply:
from the files being scanned I guess

Reply:
As far as I remember from time when I was working with coverity, it has been scanning scanned complete repositories to apply the rule. Now it depends what we want to achieve...

Reply:
those scripts can run anywhere you tell them and at any number of files or directories. In this context, we only run this on the files changed by a PR. The scanning for the whole tree happens using other tools like coverity and parasoft.

Reply:
ok, thank you

Reply:
@ppryga if I not mistaken this particular rule is for the object the current object translation, though there are other guidelines about same identifier across different objects. As I said, these scripts are just helpers to easily and fast catch some violations. Compliance check should be checked with a proper static analysis tool.

Reply:
Ok, thank you